src/pinyin.cpp

```diff
@@ -0,0 +1,513 @@
+#include "pinyin.h"
+
+#include <algorithm>
+#include <map>
+#include <string>
+#include <cuchar>
```

Comment:
`uchar.h` is not available on macOS, but I think it's also not needed as you are not using the conversion functions between UTF-8 and multi-byte char sequence.

Reply:
Yeah, it should be not needed, thank you for the advice